src/rag/chunk.py

```
@@ -0,0 +1,74 @@
+
+    def process_data(self, table_name: str, data_id: int, content_column: str = "description") -> Chunk | None:
+        try:
+            with sqlite3.connect(self.db_path) as conn:
+                conn.row_factory = sqlite3.Row
+                cursor = conn.cursor()
+                # Validate table_name to prevent SQL injection  
+                cursor.execute("SELECT name FROM sqlite_master WHERE type='table' AND name=?", (table_name,))  
+                if not cursor.fetchone():  
+                    self.logger.error(f"Table {table_name} does not exist")  
+                    return None
+                cursor.execute(f"SELECT * FROM [{table_name}] WHERE id = ?", (data_id,))
```

Ruff (0.14.14)

[error] Possible SQL injection vector through string-based query construction

(S608)